src/RedisSessionStateProvider/RedisConnectionWrapper.cs

```diff
@@ -89,41 +87,54 @@ public void UpdateExpiryTime(int timeToExpireInSeconds)
             redisConnection.Eval(updateExpiryTimeScript, keyArgs, valueArgs);
         }
 
-/*-------End of UpdateExpiryTime operation-----------------------------------------------------------------------------------------------------------------------------------------------*/
+        /*-------End of UpdateExpiryTime operation-----------------------------------------------------------------------------------------------------------------------------------------------*/
+
+        /*-------Start of Set operation-----------------------------------------------------------------------------------------------------------------------------------------------*/
 
-/*-------Start of Set operation-----------------------------------------------------------------------------------------------------------------------------------------------*/
-        
         // KEYS[1] = = data-id, internal-id
-        // ARGV[1] = last-index-in-list, ARGV[2] = session-timeout 
-        // ARGV[3..] = { data as key and value one by one }
+        // ARGV[1] = serialized session state, ARGV[2] = session-timeout
         // this order should not change LUA script depends on it
-        static readonly string setScript = (@" 
-                redis.call('HMSET', KEYS[1], unpack(ARGV, 3, ARGV[1]))
-                redis.call('EXPIRE',KEYS[1],ARGV[2]) 
-                redis.call('HMSET', KEYS[2], 'SessionTimeout', ARGV[2])
-                redis.call('EXPIRE',KEYS[2],ARGV[2]) 
+        private static readonly string setScript = (@"
+                redis.call('SET', KEYS[1], ARGV[1])
+                redis.call('EXPIRE',KEYS[1],ARGV[2])
+                redis.call('SET', KEYS[2], ARGV[2])
+                redis.call('EXPIRE',KEYS[2],ARGV[2])
                 return 1"
                 );
 
         private bool SetPrepare(ISessionStateItemCollection data, int sessionTimeout, out string[] keyArgs, out object[] valueArgs)
         {
             keyArgs = null;
             valueArgs = null;
-            if (data != null && data.Count > 0)
```

Comment:
Should we continue bailing immediately if `data` is null or empty?

Reply:
This was removed in the most recent PR commit